impeller/aiks/canvas.cc

```diff
@@ -280,6 +280,18 @@ void Canvas::ClipPath(const Path& path, Entity::ClipOperation clip_op) {
 }
 
 void Canvas::ClipRect(const Rect& rect, Entity::ClipOperation clip_op) {
+  auto& cull_rect = xformation_stack_.back().cull_rect;
+  if (clip_op == Entity::ClipOperation::kIntersect &&               //
+      cull_rect.has_value() &&                                      //
+      xformation_stack_.back().xformation.IsTranslationScaleOnly()  //
+  ) {
+    auto transformed_rect =
+        rect.TransformBounds(xformation_stack_.back().xformation);
+    if (transformed_rect.Contains(*cull_rect)) {
+      return;  // This clip will do nothing, so skip it.
+    }
```

Comment:
This is basically the same as Geometry::CoversArea right? If we use that method instead, then I imagine we could also handle ClipRRects. To do that we could give FilledPathGeometry an optional inner_rect parameter to do the computation (and false if not present)

That would handle the case where a rrect material is scrolled to entirely cover the screen.

I had a similar idea here: https://github.com/jonahwilliams/engine/commit/e3deb28680e9ebd181146689126529337ff688ea

(Though some of the cullrect math is wrong, its right in your PR though 😆 )